tests/subsys/pm/power_domain/prj.conf

```diff
@@ -2,5 +2,6 @@ CONFIG_ZTEST=y
 CONFIG_PM=y
 CONFIG_PM_DEVICE=y
 CONFIG_POWER_DOMAIN=y
+CONFIG_PM_DEVICE_POWER_DOMAIN_DYNAMIC=1
```

Comment:
```suggestion
CONFIG_PM_DEVICE_POWER_DOMAIN_DYNAMIC=y
```

Reply:
No, this breaks the logic, that symbols is the one that is used to check how many slots should be pre-allocated.

Reply:
so then I'd make the option name contain `_NUM` somewhere